### PR TITLE
ENSCORESW-3212: Backport HGNCParser

### PR DESCRIPTION
**Pull request by tgrego**
## Description

Port changes made to HGNCParser in ensembl-xref, including unit tests, back to ensembl:feature/xref_sprint.

## Use case

See ENSCORESW-3212

## Benefits

Salvage more work from the xref sprint.

## Possible Drawbacks

Increased test-suite run time.
MultiTestDB has been duplicated for misc-scripts/xref_mapping/t

## Testing

_Have you added/modified unit tests to test the changes?_

yes

_If so, do the tests pass/fail?_

pass

_Have you run the entire test suite and no regression was detected?_

yes

### CODE REVIEW

**Comment by tgrego** [comment]
All 36473 ensembl_manual source xrefs accounted for (homo_sapiens).
Also 812 lrg source xrefs now loaded, but broken on the master branch code.

File: misc-scripts/xref_mapping/XrefParser/HGNCParser.pm [hunk 1]
```diff
@@ -15,8 +15,55 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+
+  Please email comments or questions to the public Ensembl
+  developers list at <http://lists.ensembl.org/mailman/listinfo/dev>.
+
+  Questions may also be sent to the Ensembl help desk at
+  <http://www.ensembl.org/Help/Contact>.
+
+=head1 NAME
+
+Bio::EnsEMBL::Xref::Parser::HGNCParser
```
Comment (mkszuba) [inline misc-scripts/xref_mapping/XrefParser/HGNCParser.pm]: This and other occurrences of _Bio::EnsEMBL::Xref::Parser_ in POD should be changed to _XrefParser_

File: misc-scripts/xref_mapping/t/hgnc.t [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  <http://www.ensembl.org/Help/Contact>.
+
+=cut
+
+use strict;
+use warnings;
+use Test::More;
+use Test::Exception;
+use FindBin '$Bin';
+use lib "$Bin/";
```
Comment (mkszuba) [inline misc-scripts/xref_mapping/t/hgnc.t]: Same as for VGNCParser: do we really need to import any modules from $Bin?

Comment (tgrego) [reply misc-scripts/xref_mapping/t/hgnc.t]: nope, removed

**Comment by tgrego** [comment]
doing a runtests from the ensembl toplevel requires me to copy the testdb.conf file there, and I still get a 
```
-------------------- EXCEPTION --------------------
MSG: Required configuration file '/home/tgrego/src/ensembl/misc-scripts/xref_mapping/t/MultiTestDB.conf' does not exist
STACK Bio::EnsEMBL::Test::MultiTestDB::_eval_file /home/tgrego/src/ensembl-test/modules/Bio/EnsEMBL/Test/MultiTestDB.pm:237
STACK Bio::EnsEMBL::Test::MultiTestDB::get_db_conf /home/tgrego/src/ensembl-test/modules/Bio/EnsEMBL/Test/MultiTestDB.pm:105
STACK Bio::EnsEMBL::Test::MultiTestDB::db_conf /home/tgrego/src/ensembl-test/modules/Bio/EnsEMBL/Test/MultiTestDB.pm:325
STACK Bio::EnsEMBL::Test::MultiTestDB::_rebless /home/tgrego/src/ensembl-test/modules/Bio/EnsEMBL/Test/MultiTestDB.pm:202
STACK Bio::EnsEMBL::Test::MultiTestDB::new /home/tgrego/src/ensembl-test/modules/Bio/EnsEMBL/Test/MultiTestDB.pm:161
STACK toplevel ./misc-scripts/xref_mapping/t/hgnc.t:46
Date (localtime)    = Tue Sep 24 14:34:53 2019
Ensembl API version = 98
---------------------------------------------------
```

File: misc-scripts/xref_mapping/XrefParser/HGNCParser.pm [hunk 3]
```diff
@@ -15,8 +15,55 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+    RefSeq IDs
+    CCDS IDs
+    Locus specific databases
+  ]
+
+A core database adaptor is required.
+
+=head1 SYNOPSIS
+
+  my $parser = Bio::EnsEMBL::Xref::Parser::HGNCParser->new($db->dbh);
```
Comment (mkszuba) [inline misc-scripts/xref_mapping/XrefParser/HGNCParser.pm]: One more bit requiring a namespace change...

Comment (tgrego) [reply misc-scripts/xref_mapping/XrefParser/HGNCParser.pm]: bah, sorry I missed that!

File: misc-scripts/xref_mapping/t/hgnc.t [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+my @synonyms;
+my @expected_synonyms = qw( A1BG-AS A1BGAS FLJ23569 NCRNA00181 );
+while ( my $synonym_result = $synomym_rs->next ) {
+  push @synonyms, $synonym_result->synonym;
+}
+
+foreach my $expected_synonym (@expected_synonyms) {
+  my $test_synonym = shift @synonyms;
+  is( $test_synonym, $expected_synonym, "Synonym $expected_synonym of xref_id 2 is correct");
+}
```
Comment (ens-bwalts) [inline misc-scripts/xref_mapping/t/hgnc.t]: I'm not sure that resultset returns items in a guaranteed order. Might be better to sort `@expected_synonyms` and `@synonyms` before comparing (or possibly take advantage of `Xref::Schema::ResultSet::Synonym::check_synonym()`)

Comment (tgrego) [reply misc-scripts/xref_mapping/t/hgnc.t]: fair point, I've now sorted the returned list

**Pull request merged**